### ISSUE TITLE
Parsing error: Type expected

### ISSUE THREAD

**Issue by rovansteen** [dotansimha/graphql-code-generator#2619]
**Describe the bug**
Trying to upgrade `@graphql-codegen/typescript-operations` from 1.5.0 to 1.7.0 I get the following Typescript error after generating the types.
`Parsing error: Type expected`

The culprit is the following type:
```gql
export type VerifyMultiFactorDeviceAppMutation = { verifyMultiFactorDevice: (
    Pick<TotpMultiFactorDevice, 'id'>
    & { user: Pick<User, 'id'>
      & ProfileSecurityFragment
     }
  ) |  };
```

Notice the `) | };` on the last line.

**Mutation**
```jsx
const verifyMutation = gql`
  mutation VerifyMultiFactorDeviceApp($id: ID!, $token: String!) {
    verifyMultiFactorDevice(id: $id, token: $token, primary: true) {
      ... on TOTPMultiFactorDevice {
        id
        user {
          id
          ...ProfileSecurity
        }
      }
    }
  }
  ${profileSecurityFragments}
`;
```

```jsx
const profileSecurityFragments = gql`
  fragment ProfileSecurity on User {
    multi_factor
    multiFactorDevices {
      __typename
      ... on TOTPMultiFactorDevice {
        id
        type
        primary
      }
      ... on SMSMultiFactorDevice {
        id
        type
        primary
        phone
      }
    }
    sessions {
      current
      ip_address
      user_agent
      last_activity
    }
  }
`;
```

I'm assuming it's trying to generate a type for the `SMSMultiFactorDevice` union option on the `verifyMultiFactorDevice` type even though it's not requested in the response of the mutation.

**Environment:**

- OS: macOS 10.14.6 
- @graphql-codegen/cli: 1.7.0
- @graphql-codegen/typescript: 1.7.0
- @graphql-codegen/typescript-operations: ^1.7.0,
- @graphql-codegen/typescript-react-apollo: ^1.7.0,
- NodeJS: v12.6.0

**Comment by dotansimha** [comment]
@rovansteen I think this was already fixed in `master`, but not released yet. Can you please try the latest alpha version? it's `1.7.1-alpha-797953bb.87`.

**Comment by rovansteen** [comment]
Thanks @dotansimha. Updating to `1.7.1-alpha-797953bb.87` fixes this issue.

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
